src/main/java/org/truffleruby/language/objects/InitializeClassNode.java

```diff
@@ -63,6 +63,16 @@ public DynamicObject initialize(VirtualFrame frame, DynamicObject rubyClass, Dyn
         return initializeGeneralWithBlock(frame, rubyClass, superclass, block, true);
     }
 
+    @Specialization(guards = "!isRubyClass(superclass)")
+    public DynamicObject initializeNotClass(VirtualFrame frame, DynamicObject rubyClass, Object superclass, NotProvided block) {
```

Comment:
You can use the form `, Object maybeBlock` in such a case where it doesn't matter if a block is passed or not.

Reply:
The DSL checks should tell you need a `wasProvided(superclass` guard here, but it doesn't since this is not a `@CoreMethod` unfortunately.

Reply:
Fixed and fixed, thanks.